src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java

```diff
@@ -52,14 +54,21 @@ public void save(
 
         Claim updatedClaim = caseRepository.saveClaimantResponse(claim, response, authorization);
 
-        if (ACCEPTATION == response.getType()) {
-            formaliseResponseAcceptanceService.formalise(updatedClaim, (ResponseAcceptation) response, authorization);
-        }
+        formaliseResponseAcceptance(response, updatedClaim, authorization);
 
         eventProducer.createClaimantResponseEvent(updatedClaim);
         appInsights.trackEvent(getAppInsightsEvent(response), claim.getReferenceNumber());
     }
 
+    private void formaliseResponseAcceptance(ClaimantResponse claimantResponse, Claim claim, String authorization) {
+        Response response = claim.getResponse().orElseThrow(IllegalStateException::new);
+
+        if (ACCEPTATION == claimantResponse.getType()
+            && !ResponseUtils.isResponseStatesPaid(response)) {
```

Comment:
@ellie-harrison @sabahirfan @dharmendrak - Is this correct that if payment declaration is present in the case of part admit, claimant cannot formalise claimant response (either as CCJ or Settlement)? One of the AAT functional test is failing on my branch where defendant part admit with payment declaration, so I'm curious if this is correct? Happy to change test if this is the case?